src/test/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLoggerThreadSafetyTest.java

```diff
@@ -305,71 +334,93 @@ public void testConcurrentFlushAndMethodsOtherThanPutMetric() throws Interrupted
         int contextNum = groupedSink.getContexts().size();
         MetricsContext finalContext = groupedSink.getContexts().get(contextNum - 1);
         List<DimensionSet> dimensions = finalContext.getDimensions();
-        // check size
-        assertEquals(dimensions.size(), 30);
+
+        // check dimension size
+        assertEquals(dimensions.size(), 20000);
         for (DimensionSet dim : dimensions) {
             Assert.assertEquals(dim.getDimensionKeys().size(), 4); // there are 3 default dimensions
         }
-        // check content
+        // check dimension content
         Collections.sort(
                 dimensions,
                 Comparator.comparingInt(d -> Integer.parseInt(d.getDimensionValue("Dim"))));
-        for (int i = 0; i < 30; i++) {
+        for (int i = 0; i < 2000; i++) {
             Assert.assertEquals(dimensions.get(i).getDimensionValue("Dim"), String.valueOf(i));
         }
 
+        // check property
         int propertyCnt = 0;
         for (MetricsContext mc : groupedSink.getContexts()) {
-            propertyCnt += mc.getProperty("Property-30") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-31") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-32") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-33") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-34") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-35") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-36") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-37") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-38") == null ? 0 : 1;
-            propertyCnt += mc.getProperty("Property-39") == null ? 0 : 1;
-        }
-        assertEquals(propertyCnt, 10);
+            for (int i = 20000; i < 40000; i++) {
+                propertyCnt += mc.getProperty("Property-" + i) == null ? 0 : 1;
+            }
+        }
+        assertEquals(propertyCnt, 20000);
     }
 
     @After
     public void tearDown() throws Throwable {
         if (throwable != null) throw throwable;
+        throwable = null; // reset throwable to prevent repeat throwing
     }
 
     private Map<String, Object> parseRootNode(String event) throws JsonProcessingException {
         return new JsonMapper().readValue(event, new TypeReference<Map<String, Object>>() {});
     }
 
-    @SuppressWarnings("unchecked")
-    // can not parse all metrics if metric number exceeds MAX_METRICS_PER_EVENT
-    private ArrayList<MetricDefinitionCopy> parseMetrics(List<String> events)
-            throws JsonProcessingException {
-        Map<String, Object> rootNode = parseRootNode(events.get(0));
-        Map<String, Object> metadata = (Map<String, Object>) rootNode.get("_aws");
+    @Test
+    public void testParseMetrics() throws JsonProcessingException {
```

Comment:
is this test case to test `parseAllMetrics` to make sure this function that is used for assertion is correct right?

Reply:
Oh this is a helper test case that I forgot to delete. Will remove it.